src/tools/wpa/DataModel/QuicState.cs

```diff
@@ -0,0 +1,49 @@
+﻿//
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+//
+
+using System.Collections.Generic;
+
+namespace MsQuicTracing.DataModel
+{
+    public sealed class QuicState
+    {
+        public Dictionary<QuicObjectType, ulong> ObjectEventCounts { get; } = new Dictionary<QuicObjectType, ulong>();
+
+        public QuicObjectSet<QuicWorker> Workers { get; } =
+            new QuicObjectSet<QuicWorker>(QuicWorker.CreateEventId, QuicWorker.DestroyedEventId, QuicWorker.New);
+
+        private List<QuicEvent> Events = new List<QuicEvent>();
+
+        internal void AddEvent(QuicEvent quicEvent)
+        {
+            if (!ObjectEventCounts.ContainsKey(quicEvent.ObjectType))
+            {
+                ObjectEventCounts.Add(quicEvent.ObjectType, 1);
+            }
+            else
+            {
+                ObjectEventCounts[quicEvent.ObjectType]++;
```

Comment:
This is actually going to result in 3 hash lookups in this case. Since it seems like this is a hot path, do the following to bring it down to 2.

if (ObjectEventCounts.TryGetValue(quicEvent.ObjectType, out var eventCount)) 
{
   ObjectEventCounts[quicEvent.ObjectType] = eventCount + 1;
}
else
{
  ObjectEventCounts.Add(quicEvent.ObjectType, 1);
}

Reply:
I plan to remove all this code soon, so don't worry about it.